ggshield/core/git_shell.py

```diff
@@ -302,13 +308,14 @@ def tar_from_ref_and_filepaths(
 
     with tarfile.open(fileobj=tar_stream, mode="w:gz") as tar:
         for path in filepaths:
-            raw_file_content = git(["show", f"{ref}:./{path}"], cwd=wd)
 
-            if acceptation_func is not None and not (
-                acceptation_func(path, raw_file_content)
-            ):
+            def _readfile() -> str:
+                return read_git_file(ref, path, wd)
+
+            if acceptation_func is not None and not acceptation_func(path, _readfile):
                 continue
 
+            raw_file_content = _readfile()
```

Comment:
```suggestion
            raw_file_content = read_git_file(ref, path, wd)
```

If the `open_file` argument is removed from acceptation_func